src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractPackage.cs

```diff
@@ -27,36 +27,36 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
             ForegroundObject = new ForegroundThreadAffinitizedObject(componentModel.GetService<IThreadingContext>());
         }
 
-        protected void LoadComponentsInUIContextOnceSolutionFullyLoaded(CancellationToken cancellationToken)
+        protected async Task LoadComponentsInUIContextOnceSolutionFullyLoadedAsync(CancellationToken cancellationToken)
         {
             ThreadHelper.ThrowIfNotOnUIThread();
 
             if (KnownUIContexts.SolutionExistsAndFullyLoadedContext.IsActive)
             {
                 // if we are already in the right UI context, load it right away
-                LoadComponentsInUIContext(cancellationToken);
+                await LoadComponentsInUIContextAsync(cancellationToken);
             }
             else
             {
                 // load them when it is a right context.
-                KnownUIContexts.SolutionExistsAndFullyLoadedContext.UIContextChanged += OnSolutionExistsAndFullyLoadedContext;
+                KnownUIContexts.SolutionExistsAndFullyLoadedContext.UIContextChanged += OnSolutionExistsAndFullyLoadedContextAsync;
             }
         }
 
-        private void OnSolutionExistsAndFullyLoadedContext(object sender, UIContextChangedEventArgs e)
+        private async void OnSolutionExistsAndFullyLoadedContextAsync(object sender, UIContextChangedEventArgs e)
```

Comment:
❗️ No `async void`

Reply:
This is event handler so I have to use async void

Reply:
You can make the callback a non-async method which invokes the async method through `JoinableTaskFactory`, using `Run` or `RunAsync` as appropriate.